src/api/workers/publisher.js

```diff
@@ -1,27 +1,15 @@
-import Queue from 'bull';
 import publish from '../services/publish';
 import publishFacets from '../controller/api/publishFacets';
 import publishCharacteristics from '../services/publishCharacteristics';
 import publishDocuments from '../services/publishDocuments';
 import repositoryMiddleware from '../services/repositoryMiddleware';
+import { cleanWaitingJobsOfType } from '.';
 
-export const PUBLISH = 'publish';
-export const PUBLISHER_QUEUE = 'publisher';
+export const PUBLISHER = 'publisher';
 const listeners = [];
-export const publisherQueue = new Queue(
-    PUBLISHER_QUEUE,
-    process.env.REDIS_URL,
-    {
-        defaultJobOptions: {
-            removeOnComplete: 100,
-            removeOnFail: 100,
-            lifo: true,
-        },
-    },
-);
 
-publisherQueue.process(PUBLISH, (job, done) => {
-    publisherQueue.clean(100, 'wait');
+export const processPublication = (job, done) => {
+    cleanWaitingJobsOfType(PUBLISHER);
```

Comment:
No await here?

Reply:
Here, I don't need to wait, the active job is just starting, so there is no risk than one waiting job, go active before beeing cleaned.